rust/core-lib/src/annotations.rs

```diff
@@ -122,8 +167,35 @@ impl AnnotationStore {
     }
 
     /// Invalidates and removes all annotations in the range of the interval.
-    pub fn invalidate(&mut self, _interval: Interval) {
-        // todo
+    pub fn invalidate(&mut self, interval: Interval) {
+        for val in self.store.values_mut() {
+            let mut annotations: Vec<Annotations> = Vec::new();
```

Comment:
I think it's worth rethinking this. This implementation is doing a linear scan of each annotation from each plugin, and I'm wondering if there's a better way. I would probably start by having a `fn invalidate(&mut self, interval: Interval)` function on `Annotations`, and then in there would have the logic for operating on each `Spans` object.

There, what we really want to be doing is something that takes advantage of the tree structure of `Spans`.

It might be easiest to add a new method to spans, with a name like `delete_intersecting(iv)`, and ideally we wouldn't be scanning the whole structure, but instead taking the leading valid section, the trailing valid section, and gluing them together with a new blank section.

Reply:
You're right. I added `delete_intersecting` to `Spans` and cleaned up the invalidation logic in annotations.